backend/routes/chatRouter.js

```diff
@@ -1,6 +1,69 @@
 const express = require("express");
 const path = require("path");
 const chatRouter = express.Router();
+const Chat = require("../models/Chat.js");
+
+// Join a chat
+chatRouter.get("/join/:chat_id", async (req, res) => {
+	console.log("GET request recieved");
+    const chatId = req.params.chat_id;
+
+    try {
+		console.log("Attempting to find chat");
+        const chat = await Chat.findOne({ pin: chatId });
+		console.log("Chat found");
+
+        if (chat) {
+			res.json({ exists: true });
+        } else {
+            res.status(404).json({ exists: false, message: "Chatroom not found." });
+        }
+    } catch (error) {
+        console.error("Error checking chatroom existence:", error);
+        res.status(500).json({ message: "An error occurred while checking the chatroom." });
+    }
+});
+
+//Utility function to generate PIN
```

Comment:
This one needs discussion since right now we're using the chat id generated by MongoDB when create new Chat documents 
What do you guys think? @matthewsong18 @Claire-Vu